Documentation/CraftingNuGetPackages.md

```diff
@@ -0,0 +1,69 @@
+# Crafting a NuGet Package for MSB4U
+
+> MSBuildForUnity consumes .NET based packages without any modifications to the package. This documentation describes how C++, WinRT or packages with Unity-specific contents can easily be adapted to work with MSBuildForUnity.
+
+## Quickstart Summary
+
+Through the use of a [Build Targets or Props](https://docs.microsoft.com/en-us/nuget/create-packages/creating-a-package#include-msbuild-props-and-targets-in-a-package) file inside a NuGet package, we can leverage properties provided by MSBuildForUnity to appropriately select the contents to bring into the Unity project. The following snippet is taken from [Microsoft.Windows.MixedReality.DotNetWinRT](https://www.nuget.org/packages/Microsoft.Windows.MixedReality.DotNetWinRT/) NuGet package. It brings in the `Unity` folder packaged in the root of the NuGet package to the output folder inside Unity.
+
+```xml
+<Project>
+  <PropertyGroup>
+    <!--All extra resources for this package will be placed under this folder in the output directory.-->
+    <PackageDestinationFolder>$(MSBuildThisFileName)</PackageDestinationFolder>
+  </PropertyGroup>
+  
+  <ItemGroup Condition="'$(MSBuildForUnityVersion)' != ''">
+    <Content Include="$(MSBuildThisFileDirectory)..\Unity\**">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+      <!-- Don't show .meta files in Solution Explorer - it's not useful. -->
+      <Visible Condition="'%(Extension)' == '.meta'">false</Visible>
+      <Link>$(PackageDestinationFolder)\%(RecursiveDir)%(Filename)%(Extension)</Link>
+    </Content>
+  </ItemGroup>
+</Project>
+```
+
+## Understanding MSBuildForUnity Behavior
+
+MSBuildForUnity is based on MSBuild and thereby leverages the build rules including package and project reference resoluton of MSBuild. These steps are invoked at Editor Compilation stage with a set of MSBuildForUnity provided variables. At this stage, we have the following information available:
+
+- Unity Version (ie Major.Minor)
+- Platform being targeted (UWP, Android, Standalone, etc).
+- .NET Target Framework (4.6, .NET Standard 2.0)
+- Compilation define symbols specified
+- For UWP Platform we know Min and Target UWP SDK
+
+Because this stage is not aware of the target device, and happens before we build the player output (to Android Studio, XCode or Visual Studio) we don't have the following information:
```

Comment:
I don't fully understand how the information on this line jives with the information on line 32 - i.e. this line says that we don't know anything about target device (i.e. Android Studio, XCode, Visual Studio) but, on line32 we have the platform being targeted (UWP, Android, Standalone) - it feels like there is a mapping here, is it that the mapping isn't precise enough?